api/src/main/java/io/strimzi/api/kafka/model/KafkaConnectorConfig.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode
+public class KafkaConnectorConfig implements Serializable, UnknownPropertyPreserving {
```

Comment:
Out of curiousity, when is this serialized? In the `additionalProperties` map you have `Object` as values. Is it ensured that all values are serializable, too?

Reply:
It's serialized as YAML by the user and deserialized by the operator when consuming a CR. The status part, if any, is serialized by the operator when updating status. It's never serialized using Java serialization if that's what you were driving at.

additional properties it there purely so we can round trip even unknown properties. So in practice the only way for additional properties to be set is if they're present in the CR serves by the kube apiserver. In this case I think Jackson will just map them as untyped JsonObject, JsonArray. In any case the operator itself doesn't really care and since most of the operators don't update resources it doesn't make the difference. The only exception is the TO.

Reply:
> It's never serialized using Java serialization if that's what you were driving at.

Yes, that's what I meant. If it's not used, then it shouldn't have to implement `Serializable`.

Reply:
I agree with you, but IIRC _something_ at _some point_ required these classes to be Serializable. I thought that might have been Jackson, but I can't find this as being a thing with Jackson, so maybe it was something else. We should investigate and remove `Serializable` consistently for all `api` classes via a separate PR if it's not needed.